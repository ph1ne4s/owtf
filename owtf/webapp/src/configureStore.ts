/**
 * Create the store with dynamic reducers
 */

import { createStore, applyMiddleware, compose } from "redux";
import { fromJS } from "immutable";
import { routerMiddleware } from "react-router-redux";
import createSagaMiddleware from "redux-saga";
import { createLogger } from "redux-logger";
import createReducer from "./reducers";
import rootSaga from "./sagas";

const sagaMiddleware = createSagaMiddleware();

export default function configureStore(initialState = {}, history) {
  // Create the store with two middlewares
  // 1. sagaMiddleware: Makes redux-sagas work
  // 2. routerMiddleware: Syncs the location/URL path to the state
  const middlewares = [sagaMiddleware, routerMiddleware(history)];

  // Add loggerMiddleware if using app in development mode.
  if (process.env.NODE_ENV !== "production") {
    const logger = createLogger();
    middlewares.push(logger);
  }

  const enhancers = [applyMiddleware(...middlewares)];
  /**
   * If Redux DevTools Extension is installed use it, otherwise use Redux compose
   * http://extension.remotedev.io/#usage
   * https://medium.com/@zalmoxis/improve-your-development-workflow-with-redux-devtools-extension-f0379227ff83
   */
  const composeEnhancers =
    process.env.NODE_ENV !== "production" &&
    typeof window === "object" &&
    //@ts-ignore
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
      ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
          // TODO Try to remove when `react-router-redux` is out of beta, LOCATION_CHANGE should not be fired more than once after hot reloading
          shouldHotReload: false
        })
      : compose;

  const store = createStore(
    createReducer(),
    fromJS(initialState),
    composeEnhancers(...enhancers)
  );

  sagaMiddleware.run(rootSaga);

  // Make reducers hot reloadable.
  if (module.hot) {
    module.hot.accept("./reducers", () => {
      const nextRootReducer = require("./reducers").default;
      store.replaceReducer(nextRootReducer);
    });
  }

  return store;
}
