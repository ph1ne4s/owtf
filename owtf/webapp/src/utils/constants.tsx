/*
 * OWTF UI configuration
 *
 * This file will contain global configuration.
 */

export const API_BASE_URL: string = "/api/v1/";
export const API_BASE_ISSUE_URL: string =
  "https://github.com/owtf/owtf/issues/";
