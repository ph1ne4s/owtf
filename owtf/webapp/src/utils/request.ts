/**
 *
 * Request.js
 *
 * This file contains a util functions to make network requests.
 */

import "whatwg-fetch";

/**
 * Requests a URL, returning a promise
 *
 * @param  {string} url       The URL we want to request
 * @param  {object} [options] The options we want to pass to "fetch"
 *
 * @return {object}           The response promise
 */
export default class Request {
  private _url: string;
  private _options: object;
  constructor(url: string, options: object) {
    this._url = url;
    this._options = options === undefined ? {} : options;
  }

  /**
   * Parses the Response returned by a network request
   *
   * @param  {string} type Format in which response is required.
   * @param  {object} response A response from a network request
   *
   * @return {object} The parsed data in required format from the request
   */
  static _parseResponse(type: string, response: any): any {
    if (response.status === 204 || response.status === 205) {
      return null;
    }
    return type === "json" ? response.json() : response.text();
  }

  /**
   * Checks if a network request came back fine, and throws an error if not
   *
   * @param  {object} response   A response from a network request
   *
   * @return {object|undefined} Returns either the response, or throws an error
   */
  static _checkStatus(response: any) {
    if (response.status >= 200 && response.status < 300) {
      return response;
    }

    const error: any = new Error(response.statusText);
    error.response = response;
    throw error;
  }

  /**
   * Converts parameters from GET request to append in URL.
   *
   * @param  {object} queryParams Query parameters.
   *
   * @return {string} string to append in URL.
   */

  _getQuery(queryParams: object): string {
    const arr = Object.keys(queryParams).map(
      k => `${k}=${encodeURIComponent(queryParams[k])}`
    );
    return `?${arr.join("&")}`;
  }

  /**
   * Converts post/patch/put json data to application/x-www-form-urlencoded data format.
   *
   * @param data Data in JSON format
   *
   * @return {string} form-url-encoded string
   */

  _getFormEncodedData(data: any): string {
    const formBody = [];
    for (const property in data) {
      const encodedKey = encodeURIComponent(property);
      const encodedValue = encodeURIComponent(data[property]);
      formBody.push(`${encodedKey}=${encodedValue}`);
    }
    return formBody.join("&");
  }

  /**
   * Adds default headers for request if not passed by saga.
   *
   * @param  {object} target Object in which we are adding default headers.
   * @param  {object} headers Default headers to add.
   */

  static _addDefaults(target: object, headers: object) {
    for (const prop in headers) {
      target[prop] = target[prop] || headers[prop];
    }
  }

  /**
   * Requests a URL, returning a promise
   *
   * @param  {string} method HTTP Method of request.
   * @param  {string} url URL of request.
   * @param  {object} opts optional headers.
   * @param  {object} data Data to submit. Required by POST, PATCH, DELETE, PUT requests.
   * @param  {object} queryParams Query Parameters for GET requests.
   *
   * @return {object} The response data.
   */

  _fetch(
    method: string,
    url: string,
    opts: any,
    data: any = null,
    queryParams: any = null
  ): object {
    opts.method = method;
    opts.headers = opts.headers || {};
    opts.responseAs =
      opts.responseAs && ["json", "text"].includes(opts.responseAs)
        ? opts.responseAs
        : "json";

    Request._addDefaults(opts.headers, {
      Accept: "application/json"
    });

    if (queryParams) {
      url += this._getQuery(queryParams);
    }

    if (data) {
      opts.body = this._getFormEncodedData(data);
    } else {
      delete opts.body;
    }

    return fetch(url, opts)
      .then(Request._checkStatus)
      .then(Request._parseResponse.bind(null, opts.responseAs));
  }

  /**
   * Method to make GET requests.
   *
   * @param  {object} queryParams Query Parameters for GET requests.
   *
   * @return {object} The response data.
   */

  get(queryParams: object): object {
    return this._fetch("GET", this._url, this._options, null, queryParams);
  }

  /**
   * Method to make POST requests.
   *
   * @param  {object} data Data to submit. Required by POST, PATCH, DELETE, PUT requests.
   *
   * @return {object} The response data.
   */

  post(data: object): object {
    return this._fetch("POST", this._url, this._options, data);
  }

  /**
   * Method to make PATCH requests.
   *
   * @param  {object} data Data to submit.
   *
   * @return {object} The response data.
   */

  patch(data: object): object {
    return this._fetch("PATCH", this._url, this._options, data);
  }

  /**
   * Method to make PUT requests.
   *
   * @param  {object} data Data to submit.
   *
   * @return {object} The response data.
   */

  put(data: object): object {
    return this._fetch("PUT", this._url, this._options, data);
  }

  /**
   * Method to make DELETE requests.
   *
   * @return {object} The response data.
   */

  delete(): object {
    return this._fetch("DELETE", this._url, this._options);
  }
}
