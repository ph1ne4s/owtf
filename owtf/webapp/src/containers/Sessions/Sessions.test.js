import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import "../../setupTests";
import { findByTestAtrr, checkProps } from "../../utils/testUtils";
import configureStore from "redux-mock-store";
import ConnectedSessions, { Sessions } from "./index";
import SessionTable from "./SessionTable";
import { fromJS } from "immutable";

import {
  loadSessions,
  sessionsChanged,
  sessionsChangingError,
  sessionsLoaded,
  sessionsLoadingError,
  sessionsCreated,
  sessionsCreatingError,
  sessionsDeleted,
  sessionsDeletingError
} from "./actions";
import {
  CHANGE_SESSION,
  CHANGE_SESSION_SUCCESS,
  CHANGE_SESSION_ERROR,
  LOAD_SESSIONS,
  LOAD_SESSIONS_SUCCESS,
  LOAD_SESSIONS_ERROR,
  CREATE_SESSION,
  CREATE_SESSION_SUCCESS,
  CREATE_SESSION_ERROR,
  DELETE_SESSION,
  DELETE_SESSION_SUCCESS,
  DELETE_SESSION_ERROR
} from "./constants";

import { getSessions, postSession, patchSession, deleteSession } from "./saga";
import { runSaga } from "redux-saga";
import * as api from "./api";

import {
  sessionsLoadReducer,
  sessionChangeReducer,
  sessionCreateReducer,
  sessionDeleteReducer
} from "./reducer";

const setUp = (initialState = {}) => {
  const mockStore = configureStore();
  const store = mockStore(initialState);
  const wrapper = shallow(<ConnectedSessions store={store} />);
  return wrapper;
};

describe("Sessions componemt", () => {
  describe("Testing dumb sessions component", () => {
    let wrapper;
    let props;
    beforeEach(() => {
      props = {
        loading: false,
        error: false,
        sessions: [
          {
            active: true,
            id: 1,
            name: "test session"
          }
        ],
        onFetchSession: jest.fn(),
        onCreateSession: jest.fn(),
        onChangeSession: jest.fn(),
        onDeleteSession: jest.fn()
      };
      wrapper = shallow(<Sessions {...props} />);
    });

    it("Should have correct prop-types", () => {
      const expectedProps = {
        loading: false,
        error: {},
        sessions: false,
        onFetchSession: () => {},
        onCreateSession: () => {},
        onChangeSession: () => {},
        onDeleteSession: () => {}
      };
      const propsErr = checkProps(Sessions, expectedProps);
      expect(propsErr).toBeUndefined();
    });

    it("Should render without errors", () => {
      const component = findByTestAtrr(wrapper, "sessionsComponent");
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it("Should correctly render its sub-components", () => {
      wrapper.setState({ newSessionName: "new session" });
      const textInput = wrapper.find("input");
      const dialogBox = wrapper.find("Dialog");
      const button = wrapper.find("button");
      const sessionTable = wrapper.find("SessionsTable");

      expect(textInput.length).toBe(2);
      expect(textInput.at(0).props().placeholder).toEqual("test session");
      expect(textInput.at(0).props().disabled).toEqual(true);
      expect(textInput.at(1).props().placeholder).toEqual(
        "Enter new session...."
      );
      expect(textInput.at(1).props().value).toEqual("new session");
      expect(dialogBox.length).toBe(1);
      expect(button.length).toBe(2);
      expect(button.at(0).props().children).toEqual("Session");
      expect(button.at(1).props().children).toEqual("Add!");
      expect(button.at(1).props().disabled).toEqual(false);
      expect(sessionTable.length).toBe(1);
    });

    it("Should update state on TextInput change event", () => {
      const textInput = wrapper
        .find(".sessionsContainer__newSessionContainer__input")
        .at(0);
      const event = {
        preventDefault() {},
        target: { value: "new session", name: "newSessionName" }
      };
      textInput.simulate("change", event);
      expect(wrapper.instance().state.newSessionName).toEqual("new session");
    });

    it("getCurrentSession method should return current session", () => {
      const sessionsInstance = wrapper.instance();
      const currentSession = sessionsInstance.getCurrentSession();
      expect(currentSession.name).toEqual("test session");
    });

    it("Should call onCreateSession on add session button click", () => {
      expect(props.onFetchSession.mock.calls.length).toBe(1);
      expect(props.onCreateSession.mock.calls.length).toBe(0);
      const addButton = wrapper
        .find(".sessionsContainer__newSessionContainer__button")
        .at(0);
      addButton.simulate("click");
      expect(props.onCreateSession.mock.calls.length).toBe(1);
    });
  });

  describe("Testing connected Sessions component", () => {
    let wrapper;
    let initialState;
    beforeEach(() => {
      const load = {
        loading: true,
        error: false,
        sessions: false
      };
      const change = {
        loading: false,
        error: false,
        currentSession: { id: 1, name: "test session" }
      };
      const create = {
        loading: false,
        error: false
      };
      const remove = {
        loading: false,
        error: false
      };
      const sessions = {
        load,
        change,
        create,
        delete: remove
      };
      initialState = fromJS({
        sessions
      });
      wrapper = setUp(initialState);
    });

    it("Props should match the initial state", () => {
      const sessionsProp = initialState
        .get("sessions")
        .get("load")
        .get("sessions");
      const fetchErrorProp = initialState
        .get("sessions")
        .get("load")
        .get("error");
      const fetchLoadingProp = initialState
        .get("sessions")
        .get("load")
        .get("loading");

      expect(wrapper.props().sessions).toEqual(sessionsProp);
      expect(wrapper.props().loading).toEqual(fetchLoadingProp);
      expect(wrapper.props().error).toEqual(fetchErrorProp);
    });
  });

  describe("Testing Session Table component", () => {
    let wrapper;
    let props;
    beforeEach(() => {
      props = {
        sessions: [
          {
            active: true,
            id: 1,
            name: "test session 1"
          },
          {
            active: false,
            id: 2,
            name: "test session 2"
          }
        ],
        loading: false,
        error: false,
        onChangeSession: jest.fn(),
        onDeleteSession: jest.fn()
      };
      wrapper = shallow(<SessionTable {...props} />);
    });

    it("Should have correct prop-types", () => {
      const expectedProps = {
        sessions: false,
        loading: true,
        error: false,
        onChangeSession: () => {},
        onDeleteSession: () => {}
      };
      const propsErr = checkProps(SessionTable, expectedProps);
      expect(propsErr).toBeUndefined();
    });

    it("Should render without errors", () => {
      const component = wrapper.find(".sessionTableContainer");
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it("Should correctly render table's sub-components", () => {
      const searchHeader = wrapper.find(
        ".sessionTableContainer__tableHeader input"
      );
      const textHeader = wrapper.find(
        ".sessionTableContainer__tableHeader span"
      );
      const tableRow = wrapper.find(
        ".sessionTableContainer__tableBody__rowContainer"
      );
      const textCell = wrapper.find(
        ".sessionTableContainer__tableBody__rowContainer__sessionNameCell"
      );
      const iconButton = wrapper.find(
        ".sessionTableContainer__tableBody__rowContainer__deleteButtonCell button"
      );
      const radio = wrapper.find(
        ".sessionTableContainer__tableBody__rowContainer__radioCell input"
      );

      expect(searchHeader.length).toBe(1);
      expect(searchHeader.props().placeholder).toEqual(
        "Search Session name...."
      );
      expect(textHeader.length).toBe(1);
      expect(textHeader.props().children).toEqual("Delete Session");
      expect(tableRow.length).toBe(props.sessions.length);
      expect(textCell.length).toBe(props.sessions.length);
      expect(textCell.at(0).props().children).toEqual(props.sessions[0].name);
      expect(textCell.at(1).props().children).toEqual(props.sessions[1].name);
      expect(iconButton.length).toBe(props.sessions.length);
      expect(radio.length).toBe(props.sessions.length);
      expect(radio.at(0).props().checked).toEqual(true);
      expect(radio.at(1).props().checked).toEqual(false);
    });

    it("Should call prop functions on radio change and iconButton click", () => {
      const iconButton = wrapper
        .find(
          ".sessionTableContainer__tableBody__rowContainer__deleteButtonCell button"
        )
        .at(0);
      const radio = wrapper
        .find(
          ".sessionTableContainer__tableBody__rowContainer__radioCell input"
        )
        .at(0);
      iconButton.simulate("click");
      expect(props.onDeleteSession.mock.calls.length).toBe(1);
      const event = {
        preventDefault() {},
        target: { checked: true }
      };
      radio.simulate("change", event);
      expect(props.onChangeSession.mock.calls.length).toBe(1);
    });

    it("Should filter the sessions correctly", () => {
      wrapper.setState({ searchQuery: "test session" });
      expect(
        wrapper.find(".sessionTableContainer__tableBody__rowContainer").length
      ).toBe(2);
      wrapper.setState({ searchQuery: "2" });
      expect(
        wrapper.find(".sessionTableContainer__tableBody__rowContainer").length
      ).toBe(1);
      wrapper.setState({ searchQuery: "test session 4" });
      expect(
        wrapper.find(".sessionTableContainer__tableBody__rowContainer").length
      ).toBe(0);
    });
  });

  describe("Testing the sagas", () => {
    describe("Testing getSessions saga", () => {
      it("Should load sessions in case of success", async () => {
        // we push all dispatched actions to make assertions easier
        // and our tests less brittle
        const dispatchedActions = [];

        // we don't want to perform an actual api call in our tests
        // so we will mock the fetchAPI api with jest
        // this will mutate the dependency which we may reset if other tests
        // are dependent on it
        const mockedSessions = {
          status: "success",
          data: [
            {
              active: true,
              id: 1,
              name: "test session 1"
            }
          ]
        };
        api.getSessionsAPI = jest.fn(() =>
          jest.fn(() => Promise.resolve(mockedSessions))
        );

        const fakeStore = {
          getState: () => ({ state: "test" }),
          dispatch: action => dispatchedActions.push(action)
        };

        // wait for saga to complete
        await runSaga(fakeStore, getSessions).done;

        expect(api.getSessionsAPI.mock.calls.length).toBe(1);
        expect(dispatchedActions).toContainEqual(
          sessionsLoaded(mockedSessions.data)
        );
      });

      it("Should handle sessions load errors in case of failure", async () => {
        const dispatchedActions = [];

        // we simulate an error by rejecting the promise
        // then we assert if our saga dispatched the action(s) correctly
        const error = "API server is down";
        api.getSessionsAPI = jest.fn(() =>
          jest.fn(() => Promise.reject(error))
        );

        const fakeStore = {
          getState: () => ({ state: "test" }),
          dispatch: action => dispatchedActions.push(action)
        };

        await runSaga(fakeStore, getSessions).done;

        expect(api.getSessionsAPI.mock.calls.length).toBe(1);
        expect(dispatchedActions).toContainEqual(sessionsLoadingError(error));
      });
    });

    describe("Testing postSession saga", () => {
      let fakeAction;
      beforeEach(() => {
        fakeAction = {
          type: CREATE_SESSION,
          sessionName: "test session"
        };
      });

      it("Should create a new session and load sessions in case of success", async () => {
        const dispatchedActions = [];
        api.postSessionAPI = jest.fn(() => jest.fn(() => Promise.resolve()));

        const fakeStore = {
          getState: () => ({ state: "test" }),
          dispatch: action => dispatchedActions.push(action)
        };
        await runSaga(fakeStore, postSession, fakeAction).done;

        expect(api.postSessionAPI.mock.calls.length).toBe(1);
        expect(dispatchedActions).toContainEqual(sessionsCreated());
        expect(dispatchedActions).toContainEqual(loadSessions());
      });

      it("Should handle session create error in case of failure", async () => {
        const dispatchedActions = [];
        const error = "API server is down";
        api.postSessionAPI = jest.fn(() =>
          jest.fn(() => Promise.reject(error))
        );

        const fakeStore = {
          getState: () => ({ state: "test" }),
          dispatch: action => dispatchedActions.push(action)
        };

        await runSaga(fakeStore, postSession, fakeAction).done;

        expect(api.postSessionAPI.mock.calls.length).toBe(1);
        expect(dispatchedActions).toContainEqual(sessionsCreatingError(error));
      });
    });

    describe("Testing patchSession saga", () => {
      let fakeAction;
      beforeEach(() => {
        fakeAction = {
          type: CHANGE_SESSION,
          session: {
            active: false,
            id: 1,
            name: "test session"
          }
        };
      });

      it("Should activate a session and load sessions in case of success", async () => {
        const dispatchedActions = [];
        api.patchSessionAPI = jest.fn(() => jest.fn(() => Promise.resolve()));

        const fakeStore = {
          getState: () => ({ state: "test" }),
          dispatch: action => dispatchedActions.push(action)
        };
        await runSaga(fakeStore, patchSession, fakeAction).done;

        expect(api.patchSessionAPI.mock.calls.length).toBe(1);
        expect(dispatchedActions).toContainEqual(
          sessionsChanged(fakeAction.session)
        );
        expect(dispatchedActions).toContainEqual(loadSessions());
      });

      it("Should handle session change error in case of failure", async () => {
        const dispatchedActions = [];
        const error = "API server is down";
        api.patchSessionAPI = jest.fn(() =>
          jest.fn(() => Promise.reject(error))
        );

        const fakeStore = {
          getState: () => ({ state: "test" }),
          dispatch: action => dispatchedActions.push(action)
        };

        await runSaga(fakeStore, patchSession, fakeAction).done;

        expect(api.patchSessionAPI.mock.calls.length).toBe(1);
        expect(dispatchedActions).toContainEqual(sessionsChangingError(error));
      });
    });

    describe("Testing deleteSession saga", () => {
      let fakeAction;
      beforeEach(() => {
        fakeAction = {
          type: DELETE_SESSION,
          session: {
            active: false,
            id: 1,
            name: "test session"
          }
        };
      });

      it("Should delete a session and load sessions in case of success", async () => {
        const dispatchedActions = [];
        api.deleteSessionAPI = jest.fn(() => jest.fn(() => Promise.resolve()));

        const fakeStore = {
          getState: () => ({ state: "test" }),
          dispatch: action => dispatchedActions.push(action)
        };

        await runSaga(fakeStore, deleteSession, fakeAction).done;

        expect(api.deleteSessionAPI.mock.calls.length).toBe(1);
        expect(dispatchedActions).toContainEqual(sessionsDeleted());
        expect(dispatchedActions).toContainEqual(loadSessions());
      });

      it("Should handle session deleting error in case of failure", async () => {
        const dispatchedActions = [];
        const error = "API server is down";
        api.deleteSessionAPI = jest.fn(() =>
          jest.fn(() => Promise.reject(error))
        );

        const fakeStore = {
          getState: () => ({ state: "test" }),
          dispatch: action => dispatchedActions.push(action)
        };

        await runSaga(fakeStore, deleteSession, fakeAction).done;

        expect(api.deleteSessionAPI.mock.calls.length).toBe(1);
        expect(dispatchedActions).toContainEqual(sessionsDeletingError(error));
      });
    });
  });

  describe("Testing reducers", () => {
    describe("Testing sessionsLoadReducer", () => {
      let initialState;
      beforeEach(() => {
        initialState = {
          loading: false,
          error: false,
          sessions: false
        };
      });

      it("Should return the initial state", () => {
        const newState = sessionsLoadReducer(undefined, {});
        expect(newState.toJS()).toEqual(initialState);
      });

      it("Should handle LOAD_SESSIONS", () => {
        const newState = sessionsLoadReducer(undefined, {
          type: LOAD_SESSIONS
        });
        expect(newState.toJS()).toEqual({
          loading: true,
          error: false,
          sessions: false
        });
      });

      it("Should handle LOAD_SESSIONS_SUCCESS", () => {
        const sessions = [
          {
            active: false,
            id: 1,
            name: "test session"
          }
        ];
        const newState = sessionsLoadReducer(
          fromJS({
            loading: true,
            error: true,
            sessions: false
          }),
          {
            type: LOAD_SESSIONS_SUCCESS,
            sessions
          }
        );
        expect(newState.toJS()).toEqual({
          loading: false,
          error: false,
          sessions: sessions
        });
      });

      it("Should handle LOAD_SESSIONS_ERROR", () => {
        const newState = sessionsLoadReducer(
          fromJS({
            loading: true,
            error: true,
            sessions: false
          }),
          {
            type: LOAD_SESSIONS_ERROR,
            error: "Test sessions loading error"
          }
        );
        expect(newState.toJS()).toEqual({
          loading: false,
          error: "Test sessions loading error",
          sessions: false
        });
      });
    });

    describe("Testing sessionCreateReducer", () => {
      let initialState;
      beforeEach(() => {
        initialState = {
          loading: false,
          error: false
        };
      });

      it("Should return the initial state", () => {
        const newState = sessionCreateReducer(undefined, {});
        expect(newState.toJS()).toEqual(initialState);
      });

      it("Should handle CREATE_SESSION", () => {
        const newState = sessionCreateReducer(undefined, {
          type: CREATE_SESSION,
          sessionName: "test session"
        });
        expect(newState.toJS()).toEqual({ loading: true, error: false });
      });

      it("Should handle CREATE_SESSION_SUCCESS", () => {
        const newState = sessionCreateReducer(
          fromJS({
            loading: true,
            error: true
          }),
          {
            type: CREATE_SESSION_SUCCESS
          }
        );
        expect(newState.toJS()).toEqual({
          loading: false,
          error: false
        });
      });

      it("Should handle CREATE_SESSION_ERROR", () => {
        const newState = sessionCreateReducer(
          fromJS({
            loading: true,
            error: true
          }),
          {
            type: CREATE_SESSION_ERROR,
            error: "Test session creating error"
          }
        );
        expect(newState.toJS()).toEqual({
          loading: false,
          error: "Test session creating error"
        });
      });
    });

    describe("Testing sessionChangeReducer", () => {
      let initialState;
      beforeEach(() => {
        initialState = {
          loading: false,
          error: false,
          currentSession: { id: 1, name: "default session" }
        };
      });

      it("Should return the initial state", () => {
        const newState = sessionChangeReducer(undefined, {});
        expect(newState.toJS()).toEqual(initialState);
      });

      it("Should handle CHANGE_SESSION", () => {
        const newState = sessionChangeReducer(undefined, {
          type: CHANGE_SESSION,
          session: {
            active: false,
            id: 1,
            name: "test session"
          }
        });
        expect(newState.toJS()).toEqual({
          loading: true,
          error: false,
          currentSession: false
        });
      });

      it("Should handle CHANGE_SESSION_SUCCESS", () => {
        const newState = sessionChangeReducer(
          fromJS({
            loading: true,
            error: true,
            currentSession: false
          }),
          {
            type: CHANGE_SESSION_SUCCESS,
            session: {
              active: false,
              id: 1,
              name: "test session"
            }
          }
        );
        expect(newState.toJS()).toEqual({
          loading: false,
          error: false,
          currentSession: {
            active: false,
            id: 1,
            name: "test session"
          }
        });
      });

      it("Should handle CHANGE_SESSION_ERROR", () => {
        const newState = sessionChangeReducer(
          fromJS({
            loading: true,
            error: true,
            currentSession: false
          }),
          {
            type: CHANGE_SESSION_ERROR,
            error: "Test changing error"
          }
        );
        expect(newState.toJS()).toEqual({
          loading: false,
          error: "Test changing error",
          currentSession: false
        });
      });
    });

    describe("Testing sessionDeleteReducer", () => {
      let initialState;
      beforeEach(() => {
        initialState = {
          loading: false,
          error: false
        };
      });

      it("Should return the initial state", () => {
        const newState = sessionDeleteReducer(undefined, {});
        expect(newState.toJS()).toEqual(initialState);
      });

      it("Should handle DELETE_SESSION", () => {
        const newState = sessionDeleteReducer(undefined, {
          type: DELETE_SESSION,
          session: {
            active: false,
            id: 1,
            name: "test session"
          }
        });
        expect(newState.toJS()).toEqual({ loading: true, error: false });
      });

      it("Should handle DELETE_SESSION_SUCCESS", () => {
        const newState = sessionDeleteReducer(
          fromJS({
            loading: true,
            error: true
          }),
          {
            type: DELETE_SESSION_SUCCESS
          }
        );
        expect(newState.toJS()).toEqual({
          loading: false,
          error: false
        });
      });

      it("Should handle DELETE_SESSION_ERROR", () => {
        const newState = sessionDeleteReducer(
          fromJS({
            loading: true,
            error: true
          }),
          {
            type: DELETE_SESSION_ERROR,
            error: "Test deleting error"
          }
        );
        expect(newState.toJS()).toEqual({
          loading: false,
          error: "Test deleting error"
        });
      });
    });
  });
});
