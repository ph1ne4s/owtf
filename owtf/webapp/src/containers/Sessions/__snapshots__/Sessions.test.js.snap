// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sessions componemt Testing Session Table component Should render without errors 1`] = `
<div
  className="sessionTableContainer"
>
  <div
    className="sessionTableContainer__tableHeader"
  >
    <label />
    <input
      onChange={[Function]}
      placeholder="Search Session name...."
      type="text"
      value=""
    />
    <span>
      Delete Session
    </span>
  </div>
  <div
    className="sessionTableContainer__tableBody"
  >
    <div
      className="sessionTableContainer__tableBody__rowContainer"
      key="1"
    >
      <div
        className="sessionTableContainer__tableBody__rowContainer__radioCell"
      >
        <input
          checked={true}
          onChange={[Function]}
          type="radio"
        />
      </div>
      <span
        className="sessionTableContainer__tableBody__rowContainer__sessionNameCell"
      >
        test session 1
      </span>
      <div
        className="sessionTableContainer__tableBody__rowContainer__deleteButtonCell"
      >
        <button
          onClick={[Function]}
        >
          <RiDeleteBinLine />
        </button>
      </div>
    </div>
    <div
      className="sessionTableContainer__tableBody__rowContainer"
      key="2"
    >
      <div
        className="sessionTableContainer__tableBody__rowContainer__radioCell"
      >
        <input
          checked={false}
          onChange={[Function]}
          type="radio"
        />
      </div>
      <span
        className="sessionTableContainer__tableBody__rowContainer__sessionNameCell"
      >
        test session 2
      </span>
      <div
        className="sessionTableContainer__tableBody__rowContainer__deleteButtonCell"
      >
        <button
          onClick={[Function]}
        >
          <RiDeleteBinLine />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Sessions componemt Testing dumb sessions component Should render without errors 1`] = `
<div
  className="sessionsContainer"
  data-test="sessionsComponent"
>
  <input
    className="sessionsContainer__currentSessionInput"
    disabled={true}
    name="currentSession"
    placeholder="test session"
    type="text"
  />
  <button
    className="sessionsContainer__button"
    onClick={[Function]}
  >
    Session
  </button>
  <div
    className="dialogWrapper"
  >
    <Dialog
      isDialogOpened={false}
      onClose={[Function]}
      title="Sessions"
    >
      <div
        className="sessionsContainer__newSessionContainer"
      >
        <input
          className="sessionsContainer__newSessionContainer__input"
          name="newSessionName"
          onChange={[Function]}
          placeholder="Enter new session...."
          type="text"
          value=""
        />
        <button
          className="sessionsContainer__newSessionContainer__button"
          disabled={true}
          onClick={[Function]}
        >
          Add!
        </button>
      </div>
      <SessionsTable
        error={false}
        loading={false}
        onChangeSession={[MockFunction]}
        onDeleteSession={[MockFunction]}
        sessions={
          Array [
            Object {
              "active": true,
              "id": 1,
              "name": "test session",
            },
          ]
        }
      />
    </Dialog>
  </div>
</div>
`;
